### PR TITLE
feat: Add Metamask Astar Snap to native wallet options

### PR DESCRIPTION
**Pull request by gluneau**
**Pull Request Summary**

add Metamask Astar snap to native wallet options

**Check list**

- [ ] contains breaking changes
- [ ] adds new feature
- [ ] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

**This pull request makes the following changes:**

**Adds**

- (ex: Add feature A)

**Fixes**

- (ex: Fix validation function)

**Changes**

- (ex: Change document B)

**To-dos**

> \*Feel free to remove this section if it's not applicable

- [ ] (ex: add user list)

### CODE REVIEW

File: src/hooks/helper/wallet.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export const getInjectedExtensions = async (forceRequest = false): Promise<any[]
   const selectedAddress = localStorage.getItem(LOCAL_STORAGE.SELECTED_ADDRESS);
   if (selectedAddress != null || forceRequest) {
     let extensions = await web3Enable('AstarNetwork/astar-apps');
-
+    console.log('extensions', extensions);
```
Comment (impelcrypto) [inline src/hooks/helper/wallet.ts:26]: Let's remove this

File: src/modules/snap/index.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+}
+
+// export async function isPolkadotSnapInstalled(snapId: string): Promise<boolean> {
+//   // TODO: use the snapId to check if the snap is installed
+//   return !!(await getInjectedMetamaskExtension());
+// }
+
+export async function getInjectedMetamaskExtension(): Promise<InjectedMetamaskExtension | null> {
+  const extensions = await web3EnablePromise;
+  console.log('extensions', extensions);
```
Comment (impelcrypto) [inline src/modules/snap/index.ts:75]: Let's remove this

**Review by impelcrypto — APPROVED** [review]
Awesome! Let's remove lines of `console.log`.

**Comment by impelcrypto** [comment]
@bobo-k2 Please review when you are available. To test it, you have to clone [this](https://github.com/AstarNetwork/metamask-snap-astar) repo and run `yarn demo` first.

File: src/hooks/types/CustomSignature.ts [hunk 3]
```diff
@@ -29,10 +29,10 @@ export interface EcdsaAddressFormat {
...
 }
 
-declare global {
-  interface Window {
-    ethereum?: EthereumProvider;
-    talismanEth?: EthereumProvider;
-    SubWallet?: EthereumProvider;
-  }
-}
+// declare global {
```
Comment (bobo-k2) [inline src/hooks/types/CustomSignature.ts:32]: Delete if not needed

**Review by bobo-k2 — APPROVED** [review]
LGTM

**Pull request merged**